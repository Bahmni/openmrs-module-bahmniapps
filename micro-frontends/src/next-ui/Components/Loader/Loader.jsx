import React from 'react';
import { FormattedMessage } from "react-intl";
import './Loader.scss';

const Loader = () => {
    const loadingMessage = (
        <FormattedMessage
        id={"LOADING_MESSAGE"}
        defaultMessage={"Loading... Please Wait"}
        />
    );
    return (
        <div className="loader-container">
        <div className="loader-content">
            <div className="lds-spinner">
            <div></div>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
            </div>
            <p className="loader-text">{loadingMessage}</p>
        </div>
        </div>
    );
};

export default Loader;
