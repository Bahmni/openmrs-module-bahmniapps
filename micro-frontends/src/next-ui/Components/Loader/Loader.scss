.lds-spinner,
.lds-spinner div,
.lds-spinner div:after {
  box-sizing: border-box;
}
.lds-spinner {
  color: #719898;
  display: inline-block;
  position: relative;
  width: 80px;
  height: 80px;

  div {
    transform-origin: 40px 40px;
    animation: lds-spinner 1.2s linear infinite;

    &:after {
      content: " ";
      display: block;
      position: absolute;
      top: 3.2px;
      left: 36.8px;
      width: 6.4px;
      height: 17.6px;
      border-radius: 20%;
      background: #719898;
    }

    &:nth-child(1) {
      transform: rotate(0deg);
      animation-delay: -1.1s;
    }
    &:nth-child(2) {
      transform: rotate(30deg);
      animation-delay: -1s;
    }
    &:nth-child(3) {
      transform: rotate(60deg);
      animation-delay: -0.9s;
    }
    &:nth-child(4) {
      transform: rotate(90deg);
      animation-delay: -0.8s;
    }
    &:nth-child(5) {
      transform: rotate(120deg);
      animation-delay: -0.7s;
    }
    &:nth-child(6) {
      transform: rotate(150deg);
      animation-delay: -0.6s;
    }
    &:nth-child(7) {
      transform: rotate(180deg);
      animation-delay: -0.5s;
    }
    &:nth-child(8) {
      transform: rotate(210deg);
      animation-delay: -0.4s;
    }
    &:nth-child(9) {
      transform: rotate(240deg);
      animation-delay: -0.3s;
    }
    &:nth-child(10) {
      transform: rotate(270deg);
      animation-delay: -0.2s;
    }
    &:nth-child(11) {
      transform: rotate(300deg);
      animation-delay: -0.1s;
    }
    &:nth-child(12) {
      transform: rotate(330deg);
      animation-delay: 0s;
    }
  }
}

@keyframes lds-spinner {
  0% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}

.loader-container {
  font-family: "Arial";
  display: flex;
  align-items: center;
  justify-content: center;
  height: 80vh;
  width: 100vw;
  background-color: #fafafa;
}

.loader-content {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.loader-text {
  font-size: 16px;
  color: #333;
  margin: 15px 0 0;
}
