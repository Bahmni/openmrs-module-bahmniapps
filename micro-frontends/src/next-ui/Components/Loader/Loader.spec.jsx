import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import renderer from 'react-test-renderer';
import Loader from './Loader';

const messages = {
  LOADING_MESSAGE: 'Loading... Please Wait'
};

describe('Loader Component', () => {
  it('renders loader correctly', () => {
    const { getAllByText } = render(
      <IntlProvider locale="en" messages={messages}>
        <Loader />
      </IntlProvider>
    );

    expect(getAllByText("Loading... Please Wait").length).toEqual(1);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(
      <IntlProvider locale="en" messages={messages}>
        <Loader />
      </IntlProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
