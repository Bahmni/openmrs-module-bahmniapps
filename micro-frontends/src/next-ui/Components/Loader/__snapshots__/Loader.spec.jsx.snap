// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component matches snapshot 1`] = `
<div
  className="loader-container"
>
  <div
    className="loader-content"
  >
    <div
      className="lds-spinner"
    >
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
    </div>
    <p
      className="loader-text"
    >
      Loading... Please Wait
    </p>
  </div>
</div>
`;
